app/services/order_purchaser.rb

```diff
@@ -35,6 +35,12 @@ def purchase!
 
     Notifier.delay.order_receipt(user: order.user, order: order) if send_receipt?
     Notifier.delay.order_notification(order, order_notification_recipient) if send_facility_notification?
+
+    order.order_details.each do |order_detail|
```

Comment:
Would it make sense to extract all these notifications into a separate method? This `purchase!` method is getting long.